apps/prairielearn/src/ee/lib/aiQuestionGeneration.ts

```diff
@@ -230,24 +248,100 @@ A user will now request your help in in revising the question that you generated
 Keep in mind you are not just generating an example; you are generating an actual question that the user will use directly.
 `;
 
-    job.info(`system prompt is: ${sysPrompt}`);
+  job.info(`system prompt is: ${sysPrompt}`);
 
-    // TODO [very important]: normalize to prevent prompt injection attacks
+  // TODO [very important]: normalize to prevent prompt injection attacks
 
-    const completion = await client.chat.completions.create({
-      model: 'gpt-3.5-turbo',
-      messages: [
-        { role: 'system', content: sysPrompt },
-        { role: 'user', content: revisionPrompt },
-      ],
-      user: openAiUserFromAuthn(authnUserId),
-    });
+  const completion = await client.chat.completions.create({
+    model: 'gpt-3.5-turbo',
+    messages: [
+      { role: 'system', content: sysPrompt },
+      { role: 'user', content: revisionPrompt },
+    ],
+    user: openAiUserFromAuthn(authnUserId),
+  });
 
-    extractFromCompletion(completion, job);
+  extractFromCompletion(completion, job);
+  job.data['generation'] = completion.choices[0].message.content;
```

Comment:
This will overwrite the generation from the first run. I guess that's ok; we should still be logging it to the job, right?

I see that we overwrite `generation` but not `completion`. This didn't change in this PR; just calling that out for consideration.

I think all of this is fine for research-quality code. When it comes to productionizing all this, I imagine we'll want to move beyond a freeform JSON object for long-term data storage. In the meantime, it _is_ pretty convenient to not have to deal with the constraints of a database schema 🙂

Reply:
Overwriting completion for now in e49e436.